### PR TITLE
chore: moved kDrive code related to file sharing and management

### PR DESCRIPTION
**Pull request by adrien-coye**

### CODE REVIEW

File: Sources/InfomaniakCore/FileManagement/AppGroupPathProvider.swift [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+public protocol AppGroupPathProvidable: AnyObject {
+    /// Failable init if app group is not found
+    init?(appGroupIdentifier: String)
+
+    /// The directory of the current app group, exists in FS.
+    /// Uses the `.completeUntilFirstUserAuthentication` protection policy
+    var groupDirectoryURL: URL { get }
+
+    /// The root directory of kDrive, exists in FS
+    var driveRootDocumentsURL: URL { get }
```
Comment (adrien-coye) [inline Sources/InfomaniakCore/FileManagement/AppGroupPathProvider.swift:36]: discussed, use `realmRootURL`

Comment (adrien-coye) [reply Sources/InfomaniakCore/FileManagement/AppGroupPathProvider.swift]: Changed like discussed

File: Sources/InfomaniakCore/FileManagement/AppGroupPathProvider.swift [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+            return nil
+        }
+
+        do {
+            try fileManager.setAttributes(
+                [FileAttributeKey.protectionKey: FileProtectionType.completeUntilFirstUserAuthentication],
+                ofItemAtPath: groupDirectoryURL.path
+            )
+        } catch {
+            os_log("failed to protect mandatory path")
```
Comment (PhilippeWeidmann) [inline Sources/InfomaniakCore/FileManagement/AppGroupPathProvider.swift:120]: If it's already in the dependencies prefer using CocoaLumberjack

Comment (adrien-coye) [reply Sources/InfomaniakCore/FileManagement/AppGroupPathProvider.swift]: you are corect it is

**Review by PhilippeWeidmann — APPROVED** [review]
LGTM

**Pull request merged**